Step1 : 엔티티 매핑

nawhew opened the pull request:
안녕하세요. 이번 JPA 과정 코드리뷰 잘 부탁드리겠습니다 :)

nawhew committed:
feat: 기본 설정 세팅

nawhew committed:
feat: 호선(Line) 엔티티 생성

nawhew committed:
test: line 테스트 수정

nawhew committed:
feat: 역(Station) 엔티티 추가

nawhew committed:
refactor: H2DB MySQL Dialect 사용 및 도메인 GeneratedValue strategy 변경

nawhew committed:
feat: Member entity 및 repository 추가

nawhew committed:
feat: Favorite entity 및 repository 추가

nawhew committed:
refactor: 엔티티 날짜 타입 변경 Date to Timestamp

hongsii commented on src/main/java/jpa/domain/Favorite.java:
```diff
@@ -0,0 +1,42 @@
+    @Column(nullable = false)
+    private Long id;
+
+    private Timestamp created_date;
```
자바의 공식 네이밍 컨벤션은 카멜 케이스입니다. 공식 컨벤션에 따라 필드명을 바꿔 보면 어떨까요?

그리고 Java8로 넘어 가면서 날짜 기능이 개선되고 다양해진 `LocalDateTime` 을 사용을 권장하고 있습니다. 

 * https://stackoverflow.com/questions/41998144/java8-localdatetime-or-timestamp

nawhew replied:
필드명과 타입을 변경하였습니다. LocalDateTime을 알면서도 습관적으로 기존의 타입을 쓰게 됩니다 ㅠㅠ
앞으로 잘 사용하도록 하겠습니다

hongsii commented on src/main/java/jpa/domain/Favorite.java:
```diff
@@ -0,0 +1,42 @@
+
+    private Timestamp created_date;
+
+    private Timestamp modified_date;
```
JPA는 엔티티 저장 시 자동으로 생성일시와 수정일시를 관리해주는 기능이 존재합니다. 이 기능을 찾아서 적용해보면 어떨까요?

nawhew replied:
JPA Auditing 기능을 찾아서 적용하였습니다!

hongsii commented on src/main/java/jpa/domain/Favorite.java:
```diff
@@ -0,0 +1,42 @@
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
```
객체의 응집도를 높이기 위해 가능하면 setter는 지양하는 게 좋습니다!

nawhew replied:
setter를 삭제하였습니다

hongsii commented on src/test/java/jpa/domain/FavoriteRepositoryTest.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    @Test
+    @DisplayName("이름으로 favorite 조회 테스트")
```
테스트명과 테스트 내용이 다른 것 같아요!

nawhew replied:
DisplayName부분을 알맞게 수정하였습니다

hongsii commented on src/main/java/jpa/domain/Line.java:
```diff
@@ -0,0 +1,67 @@
+    @Column(unique = true)
+    private String name;
+
+    public Line(String name) {
```
실제 애플리케이션에서 JPA를 사용하려면 기본 생성자가 필요합니다. 기본 생성자를 추가해보면 어떨까요?

* https://stackoverflow.com/questions/32082633/jpa-implementation-that-doesnt-require-no-arg-constructor

nawhew replied:
네 추가하였습니다!

hongsii reviewed (changes requested):
안녕하세요! JPA 리뷰를 맡게 된 박시홍입니다.
1단계 요구사항에 맞춰 엔티티 매핑을 잘 구현해주셨습니다! 👍 
피드백 남겼으니 확인하셔서 반영해주세요.
잘 부탁드립니다! 🙇‍♂️

nawhew committed:
refactor: 엔티티 클래스 수정

수정한 엔티티 : Favorite, Line, Member, Station
- 변수명 변경 (카멜표기법 적용)
- 날짜 컬럼 타입 변경 Timestamp to LocalDateTime
- setter 삭제
- 기본 생성자 생성

nawhew committed:
feat: 엔티티 생성 및 수정일시에 JpaAutditing 적용

hongsii commented on src/main/java/jpa/common/JpaAuditingDate.java:
```diff
@@ -0,0 +1,33 @@
+ */
+@MappedSuperclass
+@EntityListeners(AuditingEntityListener.class)
+public abstract class JpaAuditingDate {
```
👍

hongsii commented on src/main/java/jpa/domain/Favorite.java:
```diff
@@ -0,0 +1,26 @@
+    @Column(nullable = false)
+    private Long id;
+
+    public Favorite() {
```
기본생성자가 무분별하게 사용되는 것을 막기 위해 `public` 보단 접근제어자의 범위를 줄여 `protected` 로 바꾸면 어떨까요?

hongsii commented on src/main/java/jpa/domain/Line.java:
```diff
@@ -0,0 +1,47 @@
+    @Column(nullable = false)
+    private Long id;
+
+    private LocalDateTime createdDate;
+
+    private LocalDateTime modifiedDate;
```
`JpaAuditingDate` 에서 정의하셨는데, 이 필드를 남겨 두신 이유가 있을까요?

hongsii reviewed (approved):
피드백 반영을 잘 해주셨네요! 👍 
추가로 피드백 남겼으니 다음 단계에서 함께 반영해주세요 :)
